### PR TITLE
Arm64 and check Compose v2 support

### PR DESCRIPTION
**Pull request by glours**
Update samples to work in the same way from an `amd64` or an `arm64` architecture and with Compose V2

### CODE REVIEW

**Review by mat007 — APPROVED** [review]
Awesome (pun intended), thanks!

**Comment by waja** [comment]
Cool. Which of this changes it relevant for docker compose v2 support? At the first look, it seems they are all unrelated to this topic.

**Comment by glours** [comment]
@waja right, there is no specific change for Compose V2, the intent of this PR was to make sure all the samples work for `arm64` and that we don't have any specific issue when running then with Compose V2

File: react-express-mysql/docker-compose.yaml [hunk 1]
```diff
@@ -28,7 +28,10 @@ services:
     depends_on:
       - db
   db:
-    image: mysql:8.0.19
+    # We use a mariadb image which support both amd64 & arm64 architecture
+    image: mariadb:10.6.4-focal
```
Comment (aiordache) [inline react-express-mysql/docker-compose.yaml:32]: Should we add the DB healthcheck to improve a bit ? A few samples have it already 😁

File: react-express-mysql/docker-compose.yaml [hunk 2]
```diff
@@ -28,7 +28,10 @@ services:
     depends_on:
       - db
   db:
-    image: mysql:8.0.19
+    # We use a mariadb image which support both amd64 & arm64 architecture
```
Comment (aiordache) [inline react-express-mysql/docker-compose.yaml:31]: Typo `support` -> `supports` (and everywhere else this comment was added).

File: sparkjava-mysql/docker-compose.yaml [hunk 3]
```diff
@@ -6,7 +6,10 @@ services:
     secrets:
       - db-password
   db:
-    image: mysql:8.0.19
+    # We use a mariadb image which support both amd64 & arm64 architecture
+    image: mariadb:10.6.4-focal
```
Comment (aiordache) [inline sparkjava-mysql/docker-compose.yaml:10]: Healthcheck !? 😅

File: wordpress-mysql/docker-compose.yaml [hunk 4]
```diff
@@ -1,6 +1,9 @@
 services:
   db:
-    image: mysql:8.0.19
+    # We use a mariadb image which support both amd64 & arm64 architecture
+    image: mariadb:10.6.4-focal
```
Comment (aiordache) [inline wordpress-mysql/docker-compose.yaml:4]: Healthcheck !? 😅

**Review by aiordache — CHANGES_REQUESTED** [review]
@glours Thank you for taking the time to update the samples 😄

File: react-express-mysql/docker-compose.yaml (hunk 1 above)
Comment (glours) [reply react-express-mysql/docker-compose.yaml:32]: I prefer to have a dedicated PR for DB healthcheck, I saw some errors when testing the samples and I think that it'll be better to spend some time completely focus on the healthcheck

**Review by aiordache — APPROVED** [review]

**Comment by maxdehaas** [comment]
Hi all,
I keep getting a mapping being expected for the restart policy. No reference on docker itself on a mapping needed as of version x. 

**ERROR: In file './docker-compose.yml', service 'restart' must be a mapping not a string.**

I tried:
`restart: always
`
or
`restart: on-failure
`
Nothing seems to work though. Should I drop a couple of versions or is anybody of you aware of this change? Or...?

**Comment by glours** [comment]
Hello @maxdehaas 
Can you create a new issue and specify which sample doesn't work? Don't forget also specify if your using Docker Desktop and the version of Docker Compose used.
Thanks

**Pull request merged**